### PR TITLE
Extended UID Support

### PR DESCRIPTION
**Pull request by kingpulse**
This pull request adds support for demonstrating provisioning with extended UID (standard UID) methods and payloads.
 
Specifics:
- Extended UID payloads can be defined in the provisioning config. This can be done by either defining the `uid` field or the individual components for the extended uid individually: `authenticator_uid_type`, `authenticator_uid`, `node_uid_type` and `node_uid`.
- Provisioning config now contains example cases for: unsecure, secure and standard UID methods
- Exception now raised if a node does not have an encryption and authentication key defined
- System works essentially the same as previously for secured method

### CODE REVIEW

**Review by GwendalRaoul — COMMENTED** [review]
Look good to me in general but could you check the comments before final approval?

File: wirepas_provisioning_server/message.py [hunk 1]
```diff
@@ -354,6 +355,32 @@ def from_message(cls, message):
         )
 
 
+class ProvisioningExtendedUID:
```
[the remark this answers is not in the document]
Comment (kingpulse) [reply wirepas_provisioning_server/message.py]: Good thinking! An internal utility function definitely makes more sense for this. I've made this change in the latest commit.

**Comment by kingpulse** [comment]
Thank you for reviewing the PR @GwendalRaoul  :) 

I've updated with PR based on the comments you have made. 

Note: I return `bytes` with the utility function instead of `bytearray` because `bytearray` is not hashable and causes an error when adding the uid to the ProvisioningData class.

File: wirepas_provisioning_server/data.py [hunk 2]
```diff
@@ -11,6 +11,68 @@
...
+
+    return b"".join([
+        authenticator_uid_type,
+        authenticator_uid,
+        node_uid_type,
+        node_uid
+    ])
+
+
+def _generate_extended_uid(authenticator_uid_type,
```
Comment (GwendalRaoul) [inline wirepas_provisioning_server/data.py:47]: Copy/paste error?

Why having this function two times?

Comment (kingpulse) [reply wirepas_provisioning_server/data.py]: Hmm. Looks like when I did a git amend to remove an import of `ProvisioningExtendedUID` my IDE merged with remote again (not sure why) and unbeknownst to me it added a second definition.

I've updated the PR to remove the second definition.

**Review by GwendalRaoul — APPROVED** [review]
Thanks for handling all the feedbacks!

**Pull request merged**